generative_data_prep/data_prep/pipeline.py

```diff
@@ -269,8 +275,31 @@ def multiprocess_data_prep(
             )
         )
 
-    with Pool(num_workers) as p:
-        _ = p.starmap(data_prep_main, data_prep_main_args_list)
+    futures = []
+    with concurrent.futures.ProcessPoolExecutor(max_workers=num_workers) as executor:
+        futures = [executor.submit(data_prep_main_helper, args) for args in data_prep_main_args_list]
```

Comment:
Does this stall the code until it is finished?

Reply:
yep